lib/spack/spack/url.py

```diff
@@ -199,6 +201,10 @@ def parse_version_offset(path):
         # https://github.com/hpc/mpileaks/releases/download/v1.0/mpileaks-1.0.tar.gz
         (r'github.com/[^/]+/[^/]+/releases/download/v?([^/]+)/.*$', path),
 
+        # e.g., https://gitlab.kitware.com/vtk/vtk/repository/archive.tar.bz2?ref=v7.0.0
+        # e.g., https://gitlab.kitware.com/icet/icet/repository/archive.tar.bz2?ref=IceT-2.1.1
```

Comment:
> also since it is specific to gitlab, should not your regex start with 'gitlab.?

on a second thought, i guess not as you can install `gitlab` on your own domain, right? Then maybe test 
```
https://my.domain.com/icet/icet/repository/archive.tar.bz2?ref=IceT-2.1.1
```
to make sure no assumption like this is used anywhere?